core/include/actsvg/core/defs.hpp

```diff
@@ -12,20 +12,20 @@
 #include <array>
 
 namespace actsvg {
-static const std::string __l = "<";
-static const std::string __r = ">\n";
-static const std::string __el = "</";
-static const std::string __er = "/>\n";
-static const std::string __fs = "/";
-static const std::string __tab = "\t";
-static const std::string __nl = "\n";
-static const std::string __blk = " ";
-static const std::string __c = ",";
-static const std::string __d = ".";
+extern const std::string __l;
+extern const std::string __r;
+extern const std::string __el;
+extern const std::string __er;
+extern const std::string __fs;
+extern const std::string __tab;
+extern const std::string __nl;
+extern const std::string __blk;
+extern const std::string __c;
+extern const std::string __d;
```

Comment:
:thinking: I don't like `extern`. I understand the motivation, to make these variables only appear in memory once. But with `extern` that "only once" applies to the entire application. I would much rather have something that would just make these variables appear once in memory inside the `actsvg::core` library.

Do these declarations need to be in a public header at all? Could they not be moved into a header that sits in the `src/` directory?

In any case, I'd prefer to set these up like:

```c++
namespace actsvg {
struct string_constants {
    static const std::string l;
    static const std::string r;
    ...
};
}
```

With the initializations taking place in this header's corresponding `cpp` file. Then the objects would also only appear once in memory, and we still wouldn't need to introduce a global symbol for each and every one of them.

Reply:
I didn't take an extensive look at where these variables are used, but if they are in any of the templated functions (which will need to stay in header files) then these will need to be in public headers as well. But I will check.